lib/parse-client-options.js

```diff
@@ -66,7 +66,7 @@ function parseOptions (userOptions) {
   /* istanbul ignore else */
 
   if (!process.browser) {
-    clientDefaults.request.agent = getRequestAgent(userOptions)
+    clientDefaults.request.agent = getRequestAgent(defaults(userOptions, DEFAULTS))
```

Comment:
I think instead of defaulting here, I would pass two arguments, like this

```
getRequestAgent(options.baseUrl, userOptions)
```

Because `options.baseUrl` can get constructed by `.host` etc